apps/jubmoji-quest/src/pages/api/team-leaderboard/index.ts

```diff
@@ -0,0 +1,154 @@
+        teamPubKeyIndexToCount.set(
+          teamPubKeyIndex,
+          teamPubKeyIndexToCount.get(teamPubKeyIndex)! + 1
+        );
+      } else {
+        teamPubKeyIndexToCount.set(teamPubKeyIndex, 1);
+      }
+    });
+
+    return res
+      .status(200)
+      .json({ scoreMap: Object.fromEntries(teamPubKeyIndexToCount) });
+
+    // Post endpoint is used for submitting a team leaderboard proof
+    // Will return a 200 only if proof is successful, and will return parameter
+    // scoreAdded equal to the number of points added to the team's score
+  } else if (req.method === "POST") {
+    try {
+      const { questId, serializedProof } = req.body;
+
+      const quest: JubmojiQuest | null = await prisma.quest.findUnique({
+        where: { id: Number(questId) },
+        include: {
+          prerequisiteCards: {
+            select: {
+              index: true,
+            },
+          },
+          collectionCards: {
+            select: {
+              index: true,
+            },
+          },
+          powers: {
+            select: {
+              id: true,
+              name: true,
+              description: true,
+              powerType: true,
+            },
+          },
+        },
+      });
+
+      if (!quest) {
+        return res.status(500).json({ message: "Quest not found" });
+      }
+
+      if (quest.proofType !== $Enums.ProofType.TEAM_LEADERBOARD) {
+        return res.status(500).json({ message: "Invalid proof type" });
+      }
+
+      const { verified, consumedSigNullifiers } = await verifyJubmojiQuestProof(
+        {
+          config: {
+            ...quest,
+            proofParams: quest.proofParams as Prisma.JsonObject,
+          },
+          serializedProof,
+          pathToCircuits: getServerPathToCircuits(),
+        }
+      );
+      if (!verified) {
+        return res.status(500).json({ message: "Proof not verified" });
+      }
+      if (!consumedSigNullifiers) {
+        return res
+          .status(500)
+          .json({ message: "Invalid signature nullifiers" });
+      }
+      if (!areAllBigIntsDifferent(consumedSigNullifiers)) {
+        return res
+          .status(500)
+          .json({ message: "Do not provide multiple of the same signature" });
+      }
+
+      const teamLeaderboardProof = JSON.parse(
+        serializedProof
+      ) as TeamLeaderboardProof;
+      const teamCardPubKeyIndex = cardPubKeys.findIndex(
+        (card) => card.pubKeyJub === teamLeaderboardProof.teamPubKey
+      );
+      if (teamCardPubKeyIndex === -1) {
+        return res.status(500).json({ message: "Invalid team public key" });
+      }
+
+      const consumedSigNullifiersHex = consumedSigNullifiers.map(bigIntToHex);
+      const previouslyUsedSigNullifiers =
+        await prisma.teamLeaderboardNullifiers.findMany({
+          where: {
+            questId: Number(questId),
+          },
+        });
+
+      const newlyConsumedSigNullifiers = consumedSigNullifiersHex.filter(
```

Comment:
nit: Can probably do something with sets here that's much more efficient. Just add all of the `newlyConsumedSigNullifiers` to a set form of `previouslyUsedSigNullifiers` and check if they exist yet

Reply:
Oops, good catch